dbms/src/main/java/org/polypheny/db/routing/UiRoutingPageUtil.java

```diff
@@ -84,28 +95,33 @@ private static void addRoutedPlanPage( AlgNode routedNode, InformationManager qu
+        snapshot = Catalog.getInstance().getSnapshot();
         InformationGroup group = new InformationGroup( page, "Selected Placements" );
         queryAnalyzer.addGroup( group );
         InformationTable table = new InformationTable(
                 group,
-                ImmutableList.of( "Entity", "Field", "Partition (Group --> ID)", "Adapter", "Physical Name" ) );
-        if ( proposedRoutingPlan.getPhysicalPlacementsOfPartitions() != null ) {
-            proposedRoutingPlan.getPhysicalPlacementsOfPartitions().forEach( ( k, v ) -> {
-                CatalogPartition catalogPartition = Catalog.getInstance().getPartition( k );
-                CatalogPartitionGroup catalogPartitionGroup = Catalog.getInstance().getPartitionGroup( catalogPartition.partitionGroupId );
-                CatalogTable catalogTable = Catalog.getInstance().getTable( catalogPartition.tableId );
-                v.forEach( p -> {
-                    CatalogColumnPlacement catalogColumnPlacement = Catalog.getInstance().getColumnPlacement( p.left, p.right );
-                    CatalogPartitionPlacement catalogPartitionPlacement = Catalog.getInstance().getPartitionPlacement( p.left, k );
-                    CatalogColumn catalogColumn = Catalog.getInstance().getColumn( catalogColumnPlacement.columnId );
-                    table.addRow(
-                            catalogTable.getNamespaceName() + "." + catalogTable.name,
-                            catalogColumn.name,
-                            catalogPartitionGroup.partitionGroupName + " --> " + catalogPartition.id,
-                            catalogPartitionPlacement.adapterUniqueName,
-                            catalogColumnPlacement.physicalSchemaName + "." + catalogPartitionPlacement.physicalTableName + "." + catalogColumnPlacement.physicalColumnName );
-                } );
-            } );
+                ImmutableList.of( "Entity", "Placement Id", "Adapter", "Allocation Id" ) );
+        if ( proposedRoutingPlan.getRoutedDistribution() != null ) {
+            RoutedDistribution distribution = proposedRoutingPlan.getRoutedDistribution();
+            LogicalEntity entity = distribution.entity();
+
+            for ( FullPartition partition : distribution.partitions() ) {
+                if ( entity.unwrap( LogicalTable.class ).isPresent() ) {
+                    for ( PartialPartition partial : partition.partials() ) {
+                        table.addRow(
+                                entity.getNamespaceName() + "." + entity.name,
+                                partial.entity().placementId,
+                                partial.entity().adapterId,
+                                partial.entity().id );
+                    }
+
+                } else if ( entity.unwrap( LogicalCollection.class ).isPresent() ) {
+                    throw new GenericRuntimeException( "Not supported" );
```

Comment:
This is a bit drastic. Can you either implement this or make it a log message. Aborting everything because the UI cannot display some debug information is a bit overkill.

Reply:
Replaced.